Duplicati/Library/Main/Operation/Backup/DataBlockProcessor.cs

```diff
@@ -95,12 +93,24 @@ public static Task Run(BackupDatabase database, Options options, ITaskReader tas
                                 //When uploading a new volume, we register the volumes and then flush the transaction
                                 // this ensures that the local database and remote storage are as closely related as possible
                                 await database.UpdateRemoteVolumeAsync(blockvolume.RemoteFilename, RemoteVolumeState.Uploading, -1, null);
-                            
+
                                 blockvolume.Close();
 
                                 await database.CommitTransactionAsync("CommitAddBlockToOutputFlush");
 
-                                await self.Output.WriteAsync(new VolumeUploadRequest(blockvolume, indexvolume));
+                                FileEntryItem blockEntry = CreateFileEntryForUpload(blockvolume, options);
+
+                                IndexVolumeWriter indexVolume = null;

```

Comment:
Minor point, but given that there is already a local variable `indexvolume`, can we rename this (perhaps to `indexVolumeWriter`)?  Having two local variables that differ only in case can make this difficult to read.

Reply:
I agree, it is confusing. It slipped past me.. good catch.